lib/health_monitor/providers/sidekiq.rb

```diff
@@ -3,23 +3,34 @@
 
 module HealthMonitor
   module Providers
-    class SidekiqException < StandardError; end
-
+    class SidekiqException < HealthError; end
+    class LatencyError < HealthError; end
+    class LatencyWarning < HealthWarning; end
     class Sidekiq < Base
       class Configuration
         DEFAULT_LATENCY_TIMEOUT = 30
 
-        attr_accessor :latency
+        attr_accessor :error_latency, :warning_latency, :latency
 
         def initialize
           @latency = DEFAULT_LATENCY_TIMEOUT
         end
+
+        def error_latency
```

Comment:
You shouldn't override a `attr_accessor` this way. Better to just provide default values during `initialize`.